optimistix/_solver/bfgs.py

```diff
@@ -216,7 +216,7 @@ def step(
         )
 
         def accepted(descent_state):
-            (grad,) = lin_to_grad(lin_fn, y)
+            (grad,) = lin_to_grad(lin_fn, tree_full_like(y, 0))
```

Comment:
I feel like passing zeros here is equally confusing as your first point, to be honest. I'd probably either just make this `state.y_eval` and/or add a comment that the value here doesn't matter.

Reply:
Alright! I now do both.

Reply:
Wonderful stuff! This LGTM.